File: controller/e_static_color.cpp
//
// Created by Nico Wittig on 2019-02-07.
//

#include "e_static_color.hpp"

e_static_color::e_static_color(light_element *le, CHSV color) : effect(le, EFFECT_NAME) {
    this->begin_color = color;
    this->end_color = color;
}

e_static_color::e_static_color(light_element *le, CHSV begin_color, CHSV end_color) : effect(le, EFFECT_NAME) {
    this->begin_color = begin_color;
    this->end_color = end_color;
}

e_static_color::e_static_color(light_element *le, CHSV begin_color, CHSV end_color, bool hue_cycle_forward) : effect(le, EFFECT_NAME) {
    this->begin_color = begin_color;
    this->end_color = end_color;
    this->hue_cycle_forward = hue_cycle_forward;
}

e_static_color::e_static_color(light_element *le, CHSV begin_color, CHSV end_color, bool hue_cycle_forward,
                               uint8_t cycles) : effect(le, EFFECT_NAME) {
    this->begin_color = begin_color;
    this->end_color = end_color;
    this->hue_cycle_forward = hue_cycle_forward;
    this->cycles = cycles;
}

e_static_color::e_static_color(light_element *le, CHSV begin_color, CHSV end_color, bool hue_cycle_forward,
                               uint8_t cycles, bool forward_only) : effect(le, EFFECT_NAME) {
    this->begin_color = begin_color;
    this->end_color = end_color;
    this->hue_cycle_forward = hue_cycle_forward;
    this->cycles = cycles;
    this->forward_only = forward_only;
}

e_static_color::~e_static_color() {
}

bool e_static_color::init() {
    set_color(begin_color, end_color, hue_cycle_forward, 0, le->get_num_leds()-1, cycles, forward_only);
    return true;
}

bool e_static_color::run() {
    return false;
}

File: server/e_static_color.py
from effect import Effect
from input import Input


class E_Static_Color(Effect):
    def __init__(self, light_element, start_color, end_color=None, hue_cycle_forward=None, cycles=1, forward_only=True, increase_hue=0, shift=0, input=None):
        super().__init__(light_element, delay=0, name="Static Color")

        self.__start_color = start_color
        self.__end_color = end_color
        self.__hue_cycle_forward = hue_cycle_forward
        self.__cycles = cycles
        self.__forward_only = forward_only
        self.__increase_hue = increase_hue
        self.__shift = shift
        self.__input = input

    def init(self):
        super().set_color(self.__start_color, self.__end_color, self.__hue_cycle_forward, cycles=self.__cycles, forward_only=self.__forward_only)
        return True

    def run(self):
        if self.__input:
            if self.__input.is_toggled():
                self.increase_hue(self.__increase_hue)
                self.shift(self.__shift)
        else:
            self.increase_hue(self.__increase_hue)
            self.shift(self.__shift)
        return True
